src/Analyzer/Resolve/IdentifierResolver.cpp

```diff
@@ -161,35 +161,29 @@ IdentifierResolveResult IdentifierResolver::tryResolveTableIdentifierFromDatabas
     storage_id = context->resolveStorageID(storage_id);
     bool is_temporary_table = storage_id.getDatabaseName() == DatabaseCatalog::TEMPORARY_DATABASE;
 
-    StoragePtr storage;
+    DatabaseAndTable database_and_table;
     TableLockHolder storage_lock;
+    std::optional<Exception> exception;
 
     if (is_temporary_table)
-        storage = DatabaseCatalog::instance().getTable(storage_id, context);
+        database_and_table = DatabaseCatalog::instance().getTableImpl(storage_id, context, &exception);
```

Comment:
What is the purpose of this change? `InterpreterSelectQuery also uses `DatabaseCatalog::tryGetTable`. I don't like the usage of `Impl` functions, and I think that it should probably be part of the public interface.